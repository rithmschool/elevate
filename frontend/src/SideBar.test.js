import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import SideBar from "./sideBar";

const routes = [
  {
    path: "/overview",
    exact: true,
    content: <h2>My Comp Overview</h2>,
    name: "My Comp Overview"
  },
  {
    path: "/offers",
    exact: true,
    content: <h2>My offers</h2>,
    name: "My offers"
  },
  {
    path: "/upload-offer",
    content: () => <h2>Upload an offer</h2>,
    name: "Upload an offer"
  }
];

// mock the route "match" prop

it("renders without crashing", function() {
  shallow(<SideBar routes={routes} />);
});

it("matches snapshot", function() {
  let wrapper = shallow(<SideBar routes={routes} />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});
