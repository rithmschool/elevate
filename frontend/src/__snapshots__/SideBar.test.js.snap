// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <Button
    color="secondary"
    onClick={[Function]}
    tag="button"
  >
    <i
      className="fas fa-th-list"
    />
  </Button>
  <div
    className="sidebar"
  >
    <ul
      className="list-group list-group-flush"
    >
      <li
        className="list-group-item"
        key="/overview"
      >
        <Link
          onClick={[Function]}
          to="/overview"
        >
          My Comp Overview
        </Link>
      </li>
      <li
        className="list-group-item"
        key="/offers"
      >
        <Link
          onClick={[Function]}
          to="/offers"
        >
          My offers
        </Link>
      </li>
      <li
        className="list-group-item"
        key="/upload-offer"
      >
        <Link
          onClick={[Function]}
          to="/upload-offer"
        >
          Upload an offer
        </Link>
      </li>
    </ul>
  </div>
</div>
`;
